### PR TITLE
Add support for up to 4 fields to multipairfield

### PR DESCRIPTION
**Pull request by sgiehl**
recreates #15125

### CODE REVIEW

**Comment by sgiehl** [comment]
@tsteur remaining UI failures are unrelated. Guess would be good to merge.

File: plugins/CoreHome/angularjs/multipairfield/multipairfield.controller.js [hunk 1]
```diff
@@ -63,6 +79,19 @@
                         hasAny = false;
                     }
                 }
+
```
Comment (tsteur) [inline plugins/CoreHome/angularjs/multipairfield/multipairfield.controller.js]: Not sure the above `else if ($scope.field2) {` is still needed? Could maybe remove the feature that only a field2 is defined but no field1? If I understand the logic correct?

BTW could slightly make above block more readable by using eg 
image

**Review by tsteur — CHANGES_REQUESTED** [review]
Left a comment. Be good to look at it. Also there's some other issue see video. You enter a value and then the other fields get the same value 

Feb-10-2020 07-59-44

File: plugins/CoreHome/angularjs/multipairfield/multipairfield.controller.js (hunk 1 above)
Comment (sgiehl) [reply plugins/CoreHome/angularjs/multipairfield/multipairfield.controller.js]: adjusted the code accordingly

**Comment by sgiehl** [comment]
> Left a comment. Be good to look at it. Also there's some other issue see video. You enter a value and then the other fields get the same value

That was because the `keys` of all fields were the same. adjusted that...

**Pull request merged**